### ISSUE TITLE
Nuclear doesn't download verified streams.

### ISSUE THREAD

**Issue by Saltane** [nukeop/nuclear#1484]
**Platform:**
Linux
**Nuclear version:**
0.6.27 (AUR)
**Description of the issue:**
Issue is quite simple. There's a track which if unverified plays undesired stream. After verifying named track to desired stream, download feature of Nuclear downloads track from undesired stream mentioned before.
**Expected behaviour:**
After verifying a track, Nuclear will download correct, verified stream.

**Comment by nukeop** [comment]
Since downloading is implemented in a different place, it does not share the code that normally checks the verification service for matches. It should be added there.

**Comment by Tengu-san** [comment]
Hello,
just a suggestion: to add a message when one add an unverified track to download, that it won't be download since this is not verified?
This will make the thing clearer :-)

**Comment by dhananjay-Byte** [comment]
@nukeop i would like to work on this issue. can you assign me this?

**Comment by nukeop** [comment]
I don't assign issues, but you're free to open a pull request.

**Comment by Krishcode264** [comment]
I've submitted a pull request #1626 that addresses the issue where Nuclear doesn't download verified streams. This enhancement allows users to select and download the desired stream directly, improving the download accuracy.

You can view the pull request [here](https://github.com/nukeop/nuclear/pull/1626) Please review and provide any feedback or suggestions.
Thank you!

**Comment by nukeop** [comment]
I decided to close that PR.

**Comment by Krishcode264** [comment]
yeah i see , we have multiple streams for single track , when user changes  the stream for perticular track from the play queue and try to download it from there , it doesnt download that newly selected stream but it always download track.streams[0] , how about if we could pass the selectedStreamId to download  service  and then it would know what perticular stream has to download for that track

**Comment by nukeop** [comment]
This issue is from an older version of Nuclear that has been replaced with a complete rewrite. None of the old issues apply to the new codebase.

The old code is preserved on the [legacy/electron](https://github.com/nukeop/nuclear/tree/legacy/electron) branch. If you have feedback about the new version, head to [Discussions](https://github.com/nukeop/nuclear/discussions).
